File: cpp/27.移除元素.cpp
/*
 * @lc app=leetcode.cn id=27 lang=cpp
 *
 * [27] 移除元素
 */
#include "leetcode.h"
// @lc code=start
class Solution
{
public:
  int removeElement(vector<int> &nums, int val)
  {
    if (nums.size() == 0) return 0;
    int slow = 0;
    int fast = 0;
    while (fast < nums.size()) {
      if (nums[fast] != val) {
        nums[slow++] = nums[fast++];
      } else {
        fast++;
      }
    }
    return slow;
  }
};
// @lc code=end

File: python/27.移除元素.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        #if not nums:return 0
        start,end = 0,len(nums)-1
        while end >=0 and nums[end]==val:
            end -=1
        while start <= end:
            if nums[start] == val:
                nums[start],nums[end] = nums[end],nums[start]
                end -= 1
            else:
                start += 1
        return start